core_types_derive/src/lib.rs

```diff
@@ -17,7 +17,10 @@ fn impl_default_json_macro(ast: &syn::DeriveInput) -> TokenStream {
             fn from(v: &#name) -> JsonString {
                 match ::serde_json::to_string(v) {
                     Ok(s) => Ok(JsonString::from(s)),
-                    Err(e) => Err(HolochainError::SerializationError(e.to_string())),
+                    Err(e) => {
+                        println!("Error serializing to JSON: {:?}", e);
+                        Err(HolochainError::SerializationError(e.to_string()))
+                    },
```

Comment:
@thedavidmeister, I've added this `println!` because I spent several hours debugging until reaching this place. For some strange reason the expect below did not lead to an output showing up, but it killed the thread this was happening in. Since it was the main redux thread that was trying to save a snapshot of the state including `pending_validations` which I had with a key that could not be serialized into a string (but that's what JSON requires) it lead to strange blocking..

Since this is hidden behind a derive, I'd rather have an error output twice rather than none, as in my case.. So I'm suggesting to add this `println` unless you have a better idea.

Reply:
`eprintln` ?

Reply:
Yes, thanks, @ddd-mtl.

Reply:
@lucksus i think that is fine, the only time we get here anyway is due to runtime data or other weird edge cases